src/Native/System.Security.Cryptography.Native/pal_ssl.cpp

```diff
@@ -565,7 +565,11 @@ extern "C" int32_t CryptoNative_SslAddExtraChainCert(SSL* ssl, X509* x509)
     }
 
     SSL_CTX* ssl_ctx = SSL_get_SSL_CTX(ssl);
-    if (SSL_CTX_add_extra_chain_cert(ssl_ctx, x509) == 1)
+	// The x509 certificate provided to SSL_CTX_add_extra_chain_cert() will be freed by the library
+	// when the SSL_CTX is destroyed. An application should not free the x509 object. Incoming X509
+	// is a disposable SafeX509Handle so we have to clone.
+	X509* clone = X509_dup(x509);
```

Comment:
The call pattern for the client code is to do this dup in managed (which you already did).  Then you need to call SetHandleAsInvalid to prevent the double-free.  As it is, this will leak certs from the client path.